### DISCUSSION TITLE
strange log

### DISCUSSION THREAD

**Discussion by bewam** [modoboa/modoboa#2177] [Help]
Hello,
With a fresh install, I can read some strange log lines:
<pre>
Feb 12 19:48:34 [TRUNKADED] dovecot: postlogin: Error: Password for user modoboa: 
Feb 12 19:48:34 [TRUNKADED] dovecot: postlogin: Error: psql: fe_sendauth: no password supplied
</pre>
I don't find where could it come from. Modoboa seems to run properly, tho.
I really don't understand.
Maybe, someone can explain me what's wrong.

**Comment by tonioo** [comment]
Looks like an issue with the postlogin script (/usr/local/bin/postlogin.sh). Looks like your postgres prevents this script from running as expected. Did you change something?

**Comment by bewam** [comment]
No, I didn't change anything in the code. 
<pre>
# This file was automatically installed on 2021-01-30T20:54:30.740493
#!/bin/sh

PATH="/usr/bin:/usr/local/bin:/bin"

psql -c "UPDATE core_user SET last_login=now() WHERE username='$USER'" > /dev/null

exec "$@"
</pre>
Maybe, would I reinstall or update?

**Reply by tonioo** [reply]
I was talking about postgres configuration.
Can you try to run the /usr/local/bin/postlogin.sh script as modoboa user?
Can you also check the /etc/dovecot/conf.d/10-master.conf file, especially the postlogin service definition, and tell what is defined user?

**Reply by bewam** [reply]
As modoboa user, /usr/local/bin/postlogin.sh
`psql: fe_sendauth: no password supplied`

I suppose, that part ?
<pre>
service postlogin {
  executable = script-login /usr/local/bin/postlogin.sh
  user = modoboa
  unix_listener postlogin {
  }
}
</pre>

Also, another line I didn't see (saying: password auth. failed in french)
<pre>
Error: pgsql(127.0.0.1): Connect failed to database modoboa: FATAL:  authentification par m
ot de passe échouée pour l'utilisateur  « modoboa »
</pre>

**Reply by jellium** [reply]
Hello @bewam did you manage to solve this issue?

**Comment by y0va** [comment]
I solved this by adding a `.pgpass` in `/srv/modoboa` like `localhost:5432:modoboa:modoboa:PASSWORD`

**Reply by jellium** [reply]
I found a `/var/lib/postgresql/.pgpass` file defined with the correct credentials. I copied it to `/srv/modoboa` (the default home directory of user `modoboa`) with the proper ownership but this did not solve this issue.

**Comment by jellium** [comment]
> Hello, With a fresh install, I can read some strange log lines:
> 
> Feb 12 19:48:34 [TRUNKADED] dovecot: postlogin: Error: Password for user modoboa: 
> Feb 12 19:48:34 [TRUNKADED] dovecot: postlogin: Error: psql: fe_sendauth: no password supplied
> 
> I don't find where could it come from. Modoboa seems to run properly, tho. I really don't understand. Maybe, someone can explain me what's wrong.

As detailed here, dovecot seems to be waiting for the password of user modoboa (as in a prompt), but gets nothing (hence the Postgresql prompt `Password for user modoboa: `) which results in connection failure.
How does the `postlogin.sh` script connects to the the psql server? Does it actually expect the passfile `.pgpass` somewhere? Or does it use the configuration of dovecot?
